plugins/partiql-plugin/src/main/kotlin/org/partiql/plugin/internal/fn/scalar/FnIsDecimal.kt

```diff
@@ -33,15 +37,25 @@ internal object FnIsDecimal1 : PartiQLFunction.Scalar {
         name = "is_decimal",
         returns = BOOL,
         parameters = listOf(
-            FunctionParameter("type_parameter_1", INT32),
-            FunctionParameter("type_parameter_2", INT32),
+            FunctionParameter("precision", INT32),
+            FunctionParameter("scale", INT32),
             FunctionParameter("value", ANY)
         ),
         isNullCall = false,
         isNullable = false,
     )
 
     override fun invoke(args: Array<PartiQLValue>): PartiQLValue {
-        TODO("Function is_decimal not implemented")
+        val p = args[0].check<Int32Value>().value
+        val s = args[1].check<Int32Value>().value
+        val v = args[2]
+        if (v !is DecimalValue) {
+            return boolValue(false)
+        }
+        val decimal = v.value
+        if (decimal == null) {
+            return boolValue(null)
+        }
+        return boolValue(decimal.precision() == p && decimal.scale() == s)
```

Comment:
`val decimal = v.value` would return a Java Big Decimal. 

The semantics of "precision" and "scale" are different in Big Decimal. 

From [Java doc](https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html): 
> The value of the number represented by the BigDecimal is therefore (unscaledValue × 10^(-scale)).
> The precision is the number of digits in the unscaled value.
>  If zero or positive, the scale is the number of digits to the right of the decimal point. If negative, the unscaled value of the number is multiplied by ten to the power of the negation of the scale.

For example: 
SQL precision 3, scale 3 <- 0.011 === 11 x 10^-3 -> precision 2, scale 3

note that depending on how you construct the big decimal, there is no guarantee on the precision and scale of the object. 

For example: 0.011 = 11 x 10^-3 = 110 x 10^-4

The safe bet was to normalize the BigDecimal, then assert on the precision and scale. 

```
                // check scale
                val decimalPoint = if (d.scale() >= 0) d.scale() else 0
                if (decimalPoint > scale) {
                    return boolValue(false)
                }
                // check integer part
                val integerPart = d.setScale(0, RoundingMode.DOWN)
                val integerLength = if (integerPart.signum() != 0) integerPart.precision() - integerPart.scale() else 0
                // PartiQL precision semantics -> the maximum number of total digit (left of decimal place + right of decimal place)
                // PartiQL scale semantics -> the total number of digit after the decimal point.
                val expectedIntegerDigits = precision - scale

                return boolValue(expectedIntegerDigits >= integerLength)
```